### PR TITLE
ROCm 3.8 Stage1 Components

### PR DESCRIPTION
**Pull request by gkathirv**

### CODE REVIEW

**Comment by gkathirv** [comment]
@adamjstewart @srekolam @arjun-raj-kuppala

File: var/spack/repos/builtin/packages/hip/package.py [hunk 1]
```diff
@@ -66,7 +67,7 @@ def patch(self):
             'hip-config.cmake.in', string=True)
 
     def flag_handler(self, name, flags):
-        if name == 'cxxflags' and '@3.7.0' in self.spec:
+        if name == 'cxxflags' and not ('@3.5.0' in self.spec):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hip/package.py]: You could also use:
```suggestion
        if name == 'cxxflags' and '@3.7.0:' in self.spec:
```
The trailing colon means an open-ended version range.

File: var/spack/repos/builtin/packages/hsa-rocr-dev/package.py [hunk 2]
```diff
@@ -14,33 +14,33 @@ class HsaRocrDev(CMakePackage):
...
+    patch('0002-Remove-explicit-RPATH-again.patch', when='@3.7.0:3.8.0')
 
     root_cmakelists_dir = 'src'
 
     def cmake_args(self):
         libelf_include = self.spec['libelf'].prefix.include.libelf
         args = ['-DLIBELF_INCLUDE_DIRS=%s' % libelf_include]
 
-        if '@3.7.0' in self.spec:
+        if not ('@3.5.0' in self.spec):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hsa-rocr-dev/package.py]: Same here

**Review by adamjstewart — APPROVED** [review]
@haampie may also be interested in reviewing this.

File: var/spack/repos/builtin/packages/rocclr/package.py [hunk 3]
```diff
@@ -24,21 +24,19 @@ def url_for_version(self, version):
         url = "https://github.com/ROCm-Developer-Tools/ROCclr/archive/rocm-{0}.tar.gz"
         return url.format(version)
 
+    version('3.8.0', sha256='10d8aa6f5af7b51813015da603c4e75edc863c3530793f6ed9769ca345c08ed6')
     version('3.7.0', sha256='a49f464bb2eab6317e87e3cc249aba3b2517a34fbdfe50175f0437f69a219adc')
     version('3.5.0', sha256='87c1ee9f02b8aa487b628c543f058198767c474cec3d21700596a73c028959e1')
 
+    depends_on('libelf', type='link')
+    depends_on('numactl', type='link')
     depends_on('cmake@3:', type='build')
```
Comment (srekolam) [inline var/spack/repos/builtin/packages/rocclr/package.py]: can you make the link dependency for numactl for 3.7.0:   That way it is not required for 3.5.0, same for libelf

File: var/spack/repos/builtin/packages/hsa-rocr-dev/package.py [hunk 4]
```diff
@@ -14,33 +14,33 @@ class HsaRocrDev(CMakePackage):
...
-    for ver in ['3.5.0', '3.7.0']:
-        depends_on('hsakmt-roct@' + ver, type=('link', 'run'), when='@' + ver)
-
-    depends_on('llvm-amdgpu@3.7.0', when='@3.7.0')
-    depends_on('libelf@0.8:', type='link')
     depends_on('cmake@3:', type="build")
+    depends_on('libelf@0.8:', type='link')
+    for ver in ['3.5.0', '3.7.0', '3.8.0']:
+        depends_on('hsakmt-roct@' + ver, type=('link', 'run'), when='@' + ver)
+        depends_on('llvm-amdgpu@' + ver, type=('link', 'run'), when='@' + ver)
```
Comment (srekolam) [inline var/spack/repos/builtin/packages/hsa-rocr-dev/package.py]: llvm-amdgpu is not required for 3.5.0. it is required from 3.7.0 onwards...

**Review by srekolam — CHANGES_REQUESTED** [review]
verify that 3.5.0 ,3.7.0 also builds

**Review by srekolam — APPROVED** [review]

File: var/spack/repos/builtin/packages/hsa-rocr-dev/package.py [hunk 5]
```diff
@@ -14,33 +14,34 @@ class HsaRocrDev(CMakePackage):
...
-    for ver in ['3.5.0', '3.7.0']:
-        depends_on('hsakmt-roct@' + ver, type=('link', 'run'), when='@' + ver)
-
-    depends_on('llvm-amdgpu@3.7.0', when='@3.7.0')
-    depends_on('libelf@0.8:', type='link')
     depends_on('cmake@3:', type="build")
+    depends_on('libelf@0.8:', type='link')
+    for ver in ['3.5.0', '3.7.0', '3.8.0']:
+        depends_on('hsakmt-roct@' + ver, type=('link', 'run'), when='@' + ver)
+        if '@3.7.0:' in ver:
```
Comment (haampie) [inline var/spack/repos/builtin/packages/hsa-rocr-dev/package.py:31]: A bit late to review here, but this looks incorrect... `ver` is just a string here

**Pull request merged**